app/routers/sessions.py

```diff
@@ -61,17 +61,17 @@ async def create_session(session: Session):
                 "max"
             ]  # ignoring min for now
 
-        # since we know that there is going to be only one question set for now
         if "question_sets" in quiz and quiz["question_sets"]:
-            question_set_id = quiz["question_sets"][0]["_id"]
-            questions = client.quiz.questions.find(
-                {"question_set_id": question_set_id}, sort=[("_id", pymongo.ASCENDING)]
-            )
-            if questions:
-                for question in questions:
+            question_ids = []
+            for question_set_index, question_set in enumerate(quiz["question_sets"]):
+                for question_index, question in enumerate(question_set["questions"]):
+                    question_ids.append(question["_id"])
+
+            if question_ids:
+                for question_id in question_ids:
                     session_answers.append(
                         jsonable_encoder(
```

Comment:
```suggestion
            question_ids = []
            for question_set_index, question_set in enumerate(quiz["question_sets"]):
                for question_index, question in enumerate(question_set["questions"]):
                    session_answers.append(
                        jsonable_encoder(
                             SessionAnswer.parse_obj({ "question_id": question["_id"] })
                        )
                    )
```

Why not do it at once?
There's no need of two separate loops.

Reply:
correct, don't know why i made two loops